arkane/logs/terachem.py

```diff
@@ -260,32 +275,49 @@ def load_scan_energies(self):
         v_list = list()
         with open(self.path, 'r') as f:
             lines = f.readlines()
-            line_index, v_index = 0, 1
-            while line_index < len(lines):
-                if 'Scan Cycle' in lines[line_index]:
-                    # example: '-=#=-     Scan Cycle     7/36        -=#=-'
+            v_index, expected_num_of_points = 0, 0
+            for line in lines:
+                if 'Scan Cycle' in line:
+                    # example: '-=#=-     Scan Cycle     5/37        -=#=-'
                     v_index += 1
-                if 'FINAL ENERGY:' in lines[line_index]:
-                    # example: 'FINAL ENERGY: -155.0341586608 a.u.'
-                    v = float(lines[line_index].split()[2])
-                    if len(v_list) == v_index:
-                        # this scan cycle still hasn't converged, overwrite the last value
-                        v_list[-1] = v
-                    else:
-                        # this is a new scan cycle, append (v_list length also increases)
+                    if not expected_num_of_points:
+                        expected_num_of_points = int(line.split()[3].split('/')[1])
+                if 'Optimized Energy:' in line:
+                    # example: '-=#=- Optimized Energy:    -155.0315243910 a.u.'
+                    v = float(line.split()[3])
+                    if len(v_list) == v_index - 1:
+                        # append this point, it is in order
                         v_list.append(v)
-                line_index += 1
+                    elif len(v_list) < v_index - 1:
+                        # seems like points in this scan are missing... add None's instead,
+                        # later they'll be removed along with the corresponding angles
+                        v_list.extend([None] * (v_index - 1 - len(v_list)))
+                    else:
+                        # we added more points that we should have, something is wrong with the log file or this method
+                        raise LogError(f'Could not parse scan energies from {self.path}')
         logging.info('   Assuming {0} is the output from a TeraChem PES scan...'.format(os.path.basename(self.path)))
 
+        v_list.append(v_list[0])  # TeraChem does not repeat the first scan point (unlike Gaussian), append it
```

Comment:
Is there a reason why we need to repeat the data at the end? I think adding data should be avoided if possible. The only reason I can think of is how we define 'angles', will have a point at the end.

A workaround might be to not add the extra datapoint and to replace the angle definition on line 310 so that it does not have the 2$\pi$ angle.

``angles = np.linspace(0, 2 * math.pi * (len(v_list) -1) / len(v_list), len(v_list), np.float64)``

Reply:
for example, if there are 4 data points (90 degree spacing), the angles output will be 0, 0.5$\pi$, $\pi$, and 1.5$\pi$.

Reply:
I found a way to repeat the last point in TeraChem instead, I'll change these lines. We do it as a quality assurance method of checking that a 360 scan gives almost the same energy as the initial point (checked in ARC)